code/modules/projectiles/guns/misc/syringe_gun.dm

```diff
@@ -102,3 +102,42 @@
 		else
 			to_chat(user, "<span class='warning'>[src] cannot hold more syringes!</span>")
 	return FALSE
+
+/obj/item/gun/syringe/chem
+	name = "reagent gun"
+	desc = "A Nanotrasen syringe gun, modified to automatically synthesise chemical darts, and instead hold reagents."
+	icon_state = "chemgun"
+	item_state = "chemgun"
+	max_syringes = 4
+	container_type = OPENCONTAINER
+	var/time_per_syringe = 250
+	var/last_synth = 0
+
+/obj/item/gun/syringe/chem/attackby(obj/item/A, mob/user, params, show_msg = TRUE)
+	return
+
+/obj/item/gun/syringe/chem/Initialize()
+	. = ..()
+	chambered = new /obj/item/ammo_casing/chemgun(src)
+	syringes.len = 4
+	START_PROCESSING(SSobj, src)
+	create_reagents(100)
+
+/obj/item/gun/syringe/chem/Destroy()
+	STOP_PROCESSING(SSobj, src)
+	return ..()
+
+/obj/item/gun/syringe/chem/process_chamber()
+	if(chambered && !chambered.BB && syringes.len)
+		chambered.newshot()
+
+/obj/item/gun/syringe/chem/process()
+	if(syringes.len >= max_syringes)

```

Comment:
Change this so instead it checks if its at max now below when it adds it and remove from the objprocess then readds when next fired.

Reply:
He's telling you to constantly turn off processing on max (as opposed to just returning) and then turn back on processing when you fire.

Is it worth it? dunno.